.gitignore

```diff
@@ -1,4 +1,8 @@
 GNUmakefile
 /dist-install/
 /dist/
+/dist-newstyle/
 ghc.mk
+*.hi
+*.o
+.ghc.environment*
```

Comment:
Fwiw, if we're adding more stuff, I'm also missing `/cabal.project.local` and possibly `/cabal.project.freeze`